src/engine.h

```diff
@@ -139,4 +141,18 @@ class EngineLoop : public UciLoop {
   EngineController engine_;
 };
 
+class Benchmark{
+ public:
+  Benchmark();
+
+  void Run();
+  void OnBestMove(const BestMoveInfo& move);
+  void OnInfo(const std::vector<ThinkingInfo>& infos);
+
+ private:
+  OptionsParser options_;
+  EngineController engine_;
```

Comment:
It doesn't seem to me that EngineController is needed at all. Even selfplay doesn't use it, why would it be needed just to populate one position and run search on it?..

But it seems that the code for creating backend and handling options --backend, --backend-opts, --weights indeed exists in 2 times already, and with benchmark it will be three, so it looks like all that functionality should be moved to `NetworkFactory` (e.g. as a helper function)

Reply:
Also if you find factoring out --backend params too complicated and wish someone else to do that, I can do that!